utils/businesslogic.py

```diff
@@ -457,21 +443,42 @@ def _service_paid_by_transaction(servicesubscription, transaction):
                 if paid_servicesubscription.state == ServiceSubscription.SUSPENDED:
                     logger.debug("Service is suspended - no action")
                 else:
-                    extra_days = timedelta(
-                        days=paid_servicesubscription.service.days_per_payment
+                    added_days = servicesubscription.paid_until - transaction.date
+                    child_days = 0
+                    # check if and howmuch forehand child servce is paid
+                    if paid_servicesubscription.paid_until:
+                        if paid_servicesubscription.paid_until > transaction.date:
+                            child_date = (
+                                paid_servicesubscription.paid_until - transaction.date
+                            )
+                            child_days = child_date.days
+
+                    extra_days = (
+                        added_days.days
+                        - servicesubscription.service.days_per_payment
+                        + paid_servicesubscription.service.days_per_payment
+                        - child_days
```

Comment:
I'd like to see a comment on the reasoning on the calculation here. It looks sane but took a while to figure it out :D